src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List getAll(Class type) {
+        MachineProducer<? extends Machine> machines;
+        if (type == Truck.class) {
+            machines = new TruckProducer();
+        } else if (type == Excavator.class) {
+            machines = new ExcavatorProducer();
+        } else if (type == Bulldozer.class) {
+            machines = new BulldozerProducer();
+        } else {
+            return new ArrayList<>();
+        }
+        return machines.get();
```

Comment:
Your code would never reach 27th line. I guess you should rethink your logic here

Reply:
+